internal/utilities/request.go

```diff
@@ -12,30 +13,55 @@ import (
 	"github.com/supabase/auth/internal/conf"
 )
 
-// GetIPAddress returns the real IP address of the HTTP request. It parses the
-// X-Forwarded-For header.
-func GetIPAddress(r *http.Request) string {
-	if r.Header != nil {
-		xForwardedFor := r.Header.Get("X-Forwarded-For")
-		if xForwardedFor != "" {
-			ips := strings.Split(xForwardedFor, ",")
-			for i := range ips {
-				ips[i] = strings.TrimSpace(ips[i])
-			}
+var (
+	ctxKeyIPAddress = &struct{}{}
+)
+
+// WithForwardedIPAddress returns a context based off the request context containing the first
+// valid IP address in the HTTP header with the given name. If no such address is found, this
+// function returns the original request context.
+//
+// Use GetIPAddress to retrieve the IP address from a request using the returned context.
+func WithForwardedIPAddress(r *http.Request, header string) context.Context {
+	var addr string
 
-			for _, ip := range ips {
-				if ip != "" {
-					parsed := net.ParseIP(ip)
-					if parsed == nil {
-						continue
-					}
+	headerValue := r.Header.Get(header)
 
-					return parsed.String()
+	if headerValue != "" {
+		ips := strings.Split(headerValue, ",")
+		for i := range ips {
+			ips[i] = strings.TrimSpace(ips[i])
+		}
+
+		for _, ip := range ips {
+			if ip != "" {
+				parsed := net.ParseIP(ip)
+				if parsed != nil {
+					addr = parsed.String()
+					break
 				}
 			}
 		}
 	}
 
+	if addr != "" {
+		ctx := context.WithValue(r.Context(), ctxKeyIPAddress, addr)
+
+		return ctx
+	}
+
+	return r.Context()
+}
+
+// GetIPAddress returns the real IP address of the HTTP request. It returns the forwarded IP address
+// in the request context if it is set. If it is not, it returns the IP address of the request
+// based on the RemoteAddr request field.
+func GetIPAddress(r *http.Request) string {
+	maybeForwardedIPAddr := r.Context().Value(ctxKeyIPAddress)
```

Comment:
Perhaps we could have a pattern like this?
if sbff.Enabled(ctx) {
    return sbff.Value(ctx)
}

Ignore the method names of course, could be sbff.IP(ctx)